### PR TITLE
added bool facet

### PR DESCRIPTION
**Pull request by frankyhollywood**
Allow for boolean facets

### CODE REVIEW

File: projects/saturn/src/test/java/io/fairspace/saturn/services/views/ViewServiceTest.java [hunk 1]
```diff
@@ -107,11 +107,5 @@ public void testFetchViewConfig() {
                 .filter(facet -> facet.getType() == ViewsConfig.ColumnType.Date)
                 .collect(Collectors.toList());
         Assert.assertEquals(2, dateFacets.size());
-
-        var boolFacets = facets.stream()
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/services/views/ViewServiceTest.java]: it looks like you added test-views.yaml introduced in the original PR, but you commented out the line, where it is used for tests. I guess this is why the tests fail for you,

File: projects/saturn/src/test/java/io/fairspace/saturn/services/views/SparqlQueryServiceTest.java [hunk 2]
```diff
@@ -110,6 +111,7 @@ public void before() throws NotAuthorizedException, BadRequestException, Conflic
         var filteredDatasetGraph = new FilteredDatasetGraph(ds.asDatasetGraph(), metadataPermissions);
         var filteredDataset = DatasetImpl.wrap(filteredDatasetGraph);
 
+//        queryService = new SparqlQueryService(ConfigLoader.CONFIG.search, loadViewsConfig("src/test/resources/test-views.yaml"), filteredDataset);
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/services/views/SparqlQueryServiceTest.java]: why did you comment out this line?

**Comment by ewelinagr** [comment]
it looks like you missed some changes from commit https://github.com/thehyve/metatree/pull/92/commits/442bf924b3c4ebb913b2dd418dd6a468f649d612
Is it on purpose?

File: projects/saturn/src/test/resources/test-views.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - name: Resource
+    title: Files
+    itemName: Name
+    types:
+      - https://sils.uva.nl/ontology#ExternalFile
+    columns:
+      - name: externalFileLink
+        title: File link
+        source: https://sils.uva.nl/ontology#externalFileLink
+        type: link
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/resources/test-views.yaml]: this is metatree specific column type, not supported in Fairspace (that is why the tests are failing). Could you remove the external file link from this config?

File: projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java [hunk 4]
```diff
@@ -27,11 +30,15 @@ public void setUp() {
 
     @Test
     public void validateVocabulary() {
-        validate(VOCABULARY, SHACL_FOR_SHACL);
+        var dsg = DatasetGraphFactory.createTxnMem();
+        Dataset ds = wrap(dsg);
+        Model model = ds.getDefaultModel();
+        var vocabulary = model.read("test-vocabulary.ttl");
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java:36]: is it possible to use the loadModel() function instead? 

```suggestion
    private static final Model TEST_VOCABULARY = loadModel("test-vocabulary.ttl");
```

File: projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java [hunk 5]
```diff
@@ -27,11 +30,15 @@ public void setUp() {
 
     @Test
     public void validateVocabulary() {
-        validate(VOCABULARY, SHACL_FOR_SHACL);
+        var dsg = DatasetGraphFactory.createTxnMem();
+        Dataset ds = wrap(dsg);
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java:34]: this one is already defined (line 24), but not used. One of these can be deleted then

File: projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java [hunk 6]
```diff
@@ -27,11 +30,15 @@ public void setUp() {
 
     @Test
     public void validateVocabulary() {
```
Comment (ewelinajozwik) [inline projects/saturn/src/test/java/io/fairspace/saturn/vocabulary/VocabulariesTest.java:32]: could the test just be 
```validate(TEST_VOCABULARY, SHACL_FOR_SHACL);```

**Pull request merged**